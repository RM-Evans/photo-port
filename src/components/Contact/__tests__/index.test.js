import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from '..';

afterEach(cleanup);

describe('contact component', () => {

    it('renders ContactForm', () => {
        render(<ContactForm />);
    });

    it('matches snapshot DOM node structure', () => {
        //render ContactForm
        const { asFragment } = render(<ContactForm />);
        //assert that snapshots match
        expect(asFragment()).toMatchSnapshot();
    });
});