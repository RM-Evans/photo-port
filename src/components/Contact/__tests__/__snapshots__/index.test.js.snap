// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section>
    <h1>
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      // name input
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      // email input
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      // message text area
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
